### ISSUE TITLE
X10S Express - can't update via Ethos Suite

### ISSUE THREAD

**Issue by aubej** [FrSkyRC/ETHOS-Feedback-Community#2113]
Using the Ethos Suite my X10S Express connects no issue, the Suite shows there is an udpate but does not show any options to update.  

Same for the Model Manager, it does not show anything in the tab.

![ethos_suite-X10s](https://user-images.githubusercontent.com/118781987/203187702-d006d294-14fb-40cb-ad4e-fc22ca7e4a0f.PNG)

**Comment by Nicholas-Luoyi** [comment]
Please try or check by following steps:
1. You could switch the server between GitHub or official server in the Settings page.
2. Please ensure your radio has a SD card and it works fine.
3. Please ensure there is a hidden file which named as sdcard.cpuid in your SD card.
4. If there isn't such a file in step 3. You should click on the SWITCH TO BOOTLOADER button.

**Comment by Nicholas-Luoyi** [comment]
After reading the code. I'm sure the Suite hasn't recognized your SD card. So it displays nothing.

**Comment by aubej** [comment]
Thank you for the quick reply.  

1. I tried the different servers with no luck.
2. I tried 3 different SD cards just in case with no luck 
- 32GB SD Ultra (the one that's usually in the radio)
- Older 2GB SD card
- Cheap 8GB Netac
All of the cards had the same result when running Etho Suite
3. The file is there (see attachment)
4. tried this anyway just in case but no luck

Thanks again for your help.


![sdcard](https://user-images.githubusercontent.com/118781987/203337028-a599b025-7f47-4ab0-8c85-87656a1e4c89.PNG)

![frserver](https://user-images.githubusercontent.com/118781987/203337096-60722072-e153-4173-ac2c-5bf1b92e434c.PNG)

**Comment by Nicholas-Luoyi** [comment]
We have tried many times and didn't meet it. I find your image looks like the one after pressing the "Eject" button. Are you sure you haven't even pressed that?

**Comment by Nicholas-Luoyi** [comment]
Oh. I have met something similar. Does your PC's file system freeze under the situation of your first image?

**Comment by aubej** [comment]
I tried again and the file system appeared to work fine.  I'm on Windows 10.  

I tried removing the sdcard.cpuid and switching to bootloader just in case.  The file got recreated but still no luck with the upgrade tools.

Thanks for the help.

**Comment by yak-54** [comment]
yep had the same issues after windows 10 system update one of the system driver has been replaced by windows
check your drivers with the radio powered on an in Frsky Suite mode and them fire up frsky suite 


![Untitled](https://user-images.githubusercontent.com/9161926/205223853-89acad16-4283-4d08-9bf3-f640df4d5290.jpg)
![Untitled2](https://user-images.githubusercontent.com/9161926/205223869-292fe642-d604-4c35-98e3-6bc8411cf74b.jpg)

**Comment by aubej** [comment]
After a bit of playing around I did get it to work.  I don't think it was the driver issue suggested above.  

My laptop (Dell XPS) only has USB-C ports so I was connected through a generic USB hub.  When I moved to the small USB-C adapter that came with the laptop the Suite worked as expected. 

![image](https://user-images.githubusercontent.com/118781987/205449694-4a1d5904-944d-41c1-a160-cfa592b998af.png)

**Comment by steffku** [comment]
> After a bit of playing around I did get it to work. I don't think it was the driver issue suggested above.
> 
> My laptop (Dell XPS) only has USB-C ports so I was connected through a generic USB hub. When I moved to the small USB-C adapter that came with the laptop the Suite worked as expected.
> 
> **It occurred to me that the generic USB hub has a micro SD card slot on it. I believe the issue is that the USB hub is seen by the OS as a storage device and when the radio is connected to the hub it's a storage device connected to a storage device which leads to the issue above.
> 
> ![image](https://user-images.githubusercontent.com/118781987/205449694-4a1d5904-944d-41c1-a160-cfa592b998af.png)

Can you post a photo or a link to the USB-C adapter, please? Maybe i have the same issue.

**Comment by aubej** [comment]
That adapter failed so I don't have it anymore.  Try connecting with an adapter that doesn't have a SD card slot on it...i.e. straight passthrough and you'll know right away if this is your issue.
